support/build/Makefile.rules

```diff
@@ -613,8 +613,8 @@ $(4): $(2) | preprocess
 		       $$(ASINCLUDES) $$(ASINCLUDES-y) \
 		       $$($(call vprefix_lib,$(1),ASINCLUDES)) $$($(call vprefix_lib,$(1),ASINCLUDES-y)) \
 		       $$($(call vprefix_src,$(1),$(2),$(3),INCLUDES)) $$($(call vprefix_src,$(1),$(2),$(3),INCLUDES-y)) \
-		       $$(ARCHFLAGS) $$(ARCHFLAGS-y) \
-		       $$(ASFLAGS) $$(ASFLAGS-y) \
+		       $$(ARCHFLAGS) $$(ARCHFLAGS-y) $$(ARCHFLAGS_EXTRA) \
```

Comment:
We shouldn't enable the ARCHFLAGS_EXTRA, the archflags are really build system specific and every compilation unit receives ARCHFLAGS, actually. We should only include language-specific extra flags.